main.c

```diff
@@ -16,20 +16,21 @@ instruction_t get_func(char *func);
  */
 int main(int argc, char *argv[])
 {
-	FILE *fd;
+	int fd, flag = 0;
 	size_t n = 0, linecount = 1, i = 0;
 	char *buff = NULL;
 	stack_t *head = NULL;
 	instruction_t code;
-	int flag = 0;
 
 	if (argc != 2)
-		dprintf(STDERR_FILENO, USAGE), exit(EXIT_FAILURE);
-	if (access(argv[1], R_OK) < 0)
-		dprintf(STDERR_FILENO, FAILOPEN, argv[1]), exit(EXIT_FAILURE);
-	fd = fopen(argv[1], "r");
+		dprintf(STDERR_FILENO, USAGE), EXIT_F;
+	fd = open(argv[1], O_RDONLY);
 	if (fd == NULL)
-		dprintf(STDERR_FILENO, FAILOPEN, argv[1]), exit(EXIT_FAILURE);
+		dprintf(STDERR_FILENO, FAILOPEN, argv[1]), EXIT_F;
+
+	if (fd = fdopen(fd, "r") == NULL)
```

Comment:
hey did you compile this? fdopen returns a FILE *, not an int. You'll need a second var, or to avoid changing everything just make a second var for open and change fd back to FILE *